components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/CacheInfo.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.proxy;
+
+import java.util.Date;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.repository.storage.Asset;
+
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.common.time.DateHelper.toDate;
+import static org.sonatype.nexus.common.time.DateHelper.toDateTime;
+
+/**
+ * A proxy cache auxiliary info for given element. {@link ProxyFacetSupport} relies on information provided by this
+ * class to implement "aging" and proxy cache invalidation.
+ *
+ * @since 3.0
+ */
+public class CacheInfo
+{
+  private static final String P_PROXY_CACHE = "proxy_cache";
```

Comment:
Maybe a sprinkle of javadoc to indicate that P_PROXY_CACHE is a map, while the others are properties? CACHE_TOKEN in particular could mean anything, but I think it's always an Etag?  ('remote cache token'?)